src/plugins/template/backend/ops/multinomial.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/reference/multinomial.hpp"
+
+#include "evaluate_node.hpp"
+
+namespace multinomial {
+
+template <ov::element::Type_t INPUT_T, ov::element::Type_t SAMPLES_T, ov::element::Type_t OUTPUT_T>
+inline void evaluate_internal(const std::shared_ptr<ov::op::v13::Multinomial>& op,
+                              const ov::HostTensorVector& outputs,
+                              const ov::HostTensorVector& inputs) {
+    using T1 = typename ov::element_type_traits<INPUT_T>::value_type;
+    using T2 = typename ov::element_type_traits<SAMPLES_T>::value_type;
+    using T3 = typename ov::element_type_traits<OUTPUT_T>::value_type;
+    ov::reference::multinomial::multinomial<T1, T2, T3>(inputs[0]->get_data_ptr<const T1>(),
```

Comment:
To fix the shape issue, try to `set_shape()` of the output tensor in the evaluate, `shape_infer` can be reused for that:
See as example: 
https://github.com/openvinotoolkit/openvino/blob/f53d880b2c9cd08b20eecee59e6e5d296c875154/src/plugins/template/backend/ops/generate_proposal.cpp#L56
https://github.com/openvinotoolkit/openvino/blob/f53d880b2c9cd08b20eecee59e6e5d296c875154/src/core/src/op/interpolate.cpp#L214-L216

Reply:
Added! Thank you so much!